Fix remaining linting errors

We've currently got ~6~ 1 linting error:

```
detector/database/cache.go:36:30: should rewrite http.NewRequestWithContext or add (*Request).WithContext (noctx)
                req, err := http.NewRequest("GET", db.ArchiveURL, nil)
                                           ^
```

We should resolve them so that we can have linting run as part of CI.

### ~`goerr113`~

~Easy to fix, I just need to remember what it's actually on about and the right fix.~

### ~`prealloc`~

~Easy to fix, but I'm concerned about the implications (though I shouldn't need to, since this is a CLI tool rather than a long-living process) because the areas its flagged are places where we don't know for sure the final length.~

~My concern is that we might over allocate, which could be worse? e.g. for npm we're deduplicating, and the number of duplicate packages could be very big or very small (so we could end up with half the size, or the same size).~

~But I also don't want to just allocate nothing to make the linter happy, as there could be a gain to be had which I think would otherwise be missed 🤔~

### `noctx`

This was me being lazy by avoiding dealing with contexts when I was initially prototyping - I want to look into this more anyway as I think it's the way to handle doing loading the OSV database and parsing any lockfiles in parallel.

--- a/detector/database/cache.go
+++ b/detector/database/cache.go
@@ -1,6 +1,7 @@
 package database
 
 import (
+	"context"
 	"encoding/json"
 	"errors"
 	"fmt"
@@ -35,7 +36,7 @@ func (db *OSVDatabase) fetchCache() (*Cache, error) {
 	}
 
 	if !db.Offline {
-		req, err := http.NewRequest("GET", db.ArchiveURL, nil)
+		req, err := http.NewRequestWithContext(context.Background(), "GET", db.ArchiveURL, nil)
 
 		if err != nil {
 			return nil, fmt.Errorf("could not retrieve OSV database archive: %w", err)

